### PR TITLE
Feature: Add the new project download service

### PR DESCRIPTION
**Pull request by Palbolsky**
## Description

This PR add the feature to download the archive of the new project.

### Add
- downloadFile service: use this service for the large file which be saved ;
- requestJson service: use this service to get a json file. It is not saved on the filesystem (currently only used in the backend, but still available in the api) ;
- checkDownloadNewProject service ;
- new-project.zip file added in the .gitignore

### Change
- the useProjectNew hook checks if an update is available and download if necessary the archive

### Delete
- new-project.zip file
- new-project.zip file removed from the package.json and forge.config.ts

## How to use it

Create a release in Github. Studio searchs always the latest release available.
In the release, 2 files are required:
- latest.json
- new-project.zip (the name can be chosen)

The latest.json file must contains:
```json
{
	"filename": "new-project.zip",
	"sha1": "8CA2287CB19604D0B8B21035BABD897539D3E5B0"
}
```
The filename is the name of the zip file in the release.
The sha1 is the signature of the zip file and can be compute with the powershell command:
`Get-FileHash .\new-project.zip -Algorithm sha1 | Format-List`

## Tests to perform

In the project creation:
- [ ] Studio download the archive
- [ ] Studio don't download the archive if it already exists and if it don't updated

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -71,6 +71,7 @@
   },
   "dependencies": {
     "@pkmn/sim": "^0.8.2",
+    "axios": "^1.6.5",
```
Comment (NuriYuri) [inline package.json:74]: Non.

Comment (NuriYuri) [reply package.json:74]: En voyant l'usage avec le stream pourquoi pas mais ça fait quand même des dépendances supplémentaires qui ne sont pas nécessaire.

File: src/backendTasks/downloadFile.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  }
+};
+
+const downloadFile = async (payload: DownloadFileInput, event: IpcMainEvent, channels: ChannelNames) => {
+  const destPath = getDestPath(payload.dest);
+  if (fs.existsSync(destPath)) {
+    fs.unlinkSync(destPath);
+  }
+  const writer = fs.createWriteStream(getDestPath(payload.dest));
+  await axios({
```
Comment (NuriYuri) [inline src/backendTasks/downloadFile.ts:34]: https://developer.mozilla.org/en-US/docs/Web/API/ReadableStream#async_iteration
https://developer.mozilla.org/en-US/docs/Web/API/Request/body

File: package.json (hunk 1 above)
Comment (Palbolsky) [reply package.json]: J'ai retiré axios. :)

File: src/backendTasks/downloadFile.ts (hunk 2 above)
Comment (Palbolsky) [reply src/backendTasks/downloadFile.ts]: Je viens seulement de voir le message. J'ai retiré axios donc l'implémentation est maintenant différente.

**Review by AerunDev — APPROVED** [review]

**Review by NuriYuri — APPROVED** [review]

**Comment by NuriYuri** [comment]
Pourquoi c'est en draft?

**Comment by Palbolsky** [comment]
> Pourquoi c'est en draft?

Parce que le repo de la demo-technique n'existe pas donc actuellement les liens de téléchargement de l'archive du new-project.zip pointent vers mon repo de test.

**Review by pw-rey — APPROVED** [review]
Tested using the very first release on the Technical Demo repository and it worked perfectly. Great job! :D

**Pull request merged**